lnd.go

```diff
@@ -667,25 +667,54 @@ func Main(cfg *Config, lisCfg ListenerCfg, implCfg *ImplementationCfg,
 	ltndLog.Infof("Waiting for chain backend to finish sync, "+
 		"start_height=%v", bestHeight)
 
+	type syncResult struct {
+		IsSynced bool
+		TS       int64
+		Err      error
+	}
+
+syncCheckLoop:
 	for {
 		if !interceptor.Alive() {
 			return nil
 		}
 
-		synced, ts, err := activeChainControl.Wallet.IsSynced()
-		if err != nil {
-			return mkErr("unable to determine if wallet is "+
-				"synced: %v", err)
-		}
+		syncedResChan := make(chan syncResult, 1)
 
-		ltndLog.Debugf("Syncing to block timestamp: %v, is synced=%v",
-			time.Unix(ts, 0), synced)
+		// We check if the wallet is synced in a separate goroutine as
+		// the call is blocking, and we want to be able to interrupt it
+		// if the daemon is shutting down.
+		go func() {
+			synced, ts, err := activeChainControl.Wallet.IsSynced()
+			syncedResChan <- syncResult{synced, ts, err}
+		}()
 
-		if synced {
-			break
+		select {
+		case <-interceptor.ShutdownChannel():
+			return nil
+
+		case res := <-syncedResChan:
+			if res.Err != nil {
+				return mkErr("unable to determine if wallet "+
+					"is synced: %v", res.Err)
+			}
+
+			ltndLog.Debugf("Syncing to block timestamp: %v, is "+
+				"synced=%v", time.Unix(res.TS, 0), res.IsSynced)
+
+			if res.IsSynced {
+				break syncCheckLoop
+			}
 		}
 
-		time.Sleep(time.Second * 1)
+		// If we're not yet synced, we'll wait for a second before
+		// checking again.
+		select {
+		case <-interceptor.ShutdownChannel():
```

Comment:
looks like we don't need to catch it again here since after 1s it will go to the next iteration.

Reply:
I still included this change, as we'll else block the shutdown for 1 second while waiting for this. If you think that's an unnecessary optimisation, I'll remove that though and just keep a:
`<-time.After(time.Second)`